pymodm/fields.py

```diff
@@ -1062,8 +1062,27 @@ def to_python(self, value):
             return self.related_model.from_document(value)
         return value
 
+    @staticmethod
+    def _value_to_mongo(related_model, value):
+        if isinstance(value, bson.SON):
+            return value
+
+        if isinstance(value, related_model):
+            return value.to_son()
+
+        # if value is a dict convert in to model
+        # so we can properly generate SON
+        if isinstance(value, dict):
+            return related_model.from_document(value).to_son()
```

Comment:
You are right if it is a dict we can't use it in mongodb as {'a': 1, 'b': 2} and {'b': 2, 'a': 1} are different values for mongodb. So it checks:
1. Is it a SON value (it has been converted already and we do not need to convert it again).
2. Is it an instance.
3. Is it a dict - then we try to create a model instance and then create a son (as model should always generate SON with the same keys order, doesn't it?)

So to_mongo will always generate SON object with the same keys order. If it is needed to check if SON is correctly converted to Model Instance too then we can combine dict and SON under one branch but again with converting to model first.